openquakeplatform/bin/oq_install.sh

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/env bash 
+# GeoNode installer script
+#
+# using getopts
+#
+
+if [ $GEM_SET_DEBUG ]; then
+    set -x
+fi
+# set -e
+
+source ~/env/bin/activate
+
+if [[ $EUID -ne 0 ]]; then
+    echo "This script must be run as root" 1>&2
+    exit 1
+fi
+
+while getopts 's:' OPTION
+do
+    case $OPTION in
+    s)
+        stepflag=1
+        stepval="$OPTARG"
+        ;;
+    ?)
+        printf "Usage: %s: [-s value] configfile\n" $(basename $0) >&2
+        exit 2
+        ;;
+    esac
+done
+shift $(($OPTIND - 1))
+
+function setup_directories() {
+    mkdir -p $GEOSERVER_DATA_DIR
+    mkdir -p $GEONODE_WWW/static
+    mkdir -p $GEONODE_WWW/uploaded
+    mkdir -p $GEONODE_WWW/wsgi
+    mkdir -p $APACHE_SITES
+    mkdir -p $GEONODE_BIN
+    mkdir -p $GEONODE_ETC
+    mkdir -p $GEONODE_ETC/media
+    mkdir -p $GEONODE_ETC/templates
+    mkdir -p $GEONODE_SHARE
+}
+
+function reorganize_configuration() {
+    cp -rp $INSTALL_DIR/support/geonode.apache $APACHE_SITES/geonode.conf
+    cp -rp $INSTALL_DIR/support/geonode.wsgi $GEONODE_WWW/wsgi/
+    cp -rp $INSTALL_DIR/support/geonode.robots $GEONODE_WWW/robots.txt
+    cp -rp $INSTALL_DIR/support/geonode.binary $GEONODE_BIN/geonode
+    cp -rp $INSTALL_DIR/GeoNode*.zip $GEONODE_SHARE
+    cp -rp $INSTALL_DIR/support/geonode.updateip $GEONODE_BIN/geonode-updateip
+    cp -rp $INSTALL_DIR/support/geonode.admin $GEONODE_SHARE/admin.json
+    cp -rp $INSTALL_DIR/support/geonode.local_settings $GEONODE_ETC/local_settings.py
+
+    chmod +x $GEONODE_BIN/geonode
+    chmod +x $GEONODE_BIN/geonode-updateip
+}
+
+function preinstall() {
+    setup_directories
+    reorganize_configuration
+    echo "Fine preinstall"
+}
+
+function randpass() {
+    [ "$2" == "0" ] && CHAR="[:alnum:]" || CHAR="[:graph:]"
+    cat /dev/urandom | tr -cd "$CHAR" | head -c ${1:-32}
+    echo
+}
+
+function setup_postgres_every_time() {
+    true # nothing to do. when we setup DB migrations they'll probably go here.
+}
+
+function setup_django_every_time() {
+    echo "setup_django_every_time:"
+    source ~/env/bin/activate
+    
+    pip -v install /usr/share/geonode/GeoNode-*.zip --no-dependencies --quiet
```

Comment:
be careful, this is executed as `root` since the script is run via sudo

Reply:
This script is in Geonode package, i have just putted other thinks for the installation